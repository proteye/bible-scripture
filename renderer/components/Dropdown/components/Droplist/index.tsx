import React, { forwardRef, MouseEvent, useCallback, useLayoutEffect, useRef } from 'react'
import mergeRefs from 'react-merge-refs'

import { IDroplistProps } from './types'

import { DROPLIST_SCROLL_DEFAULT_HEIGHT } from 'components/Dropdown/constants'
import { Scrollable } from 'components/Scrollable'

import {
  StyledChecked,
  StyledDroplist,
  StyledDroplistCancel,
  StyledDroplistHeader,
  StyledDroplistItem,
  StyledDroplistItems,
  StyledDroplistTitle,
} from './styled.index'

const Droplist = forwardRef<HTMLDivElement, IDroplistProps>(
  ({ droplistWidthAsInput, isPanel, inputRef, items, size, title, value, onSelect, close, height }, ref) => {
    const droplistRef = useRef<HTMLDivElement>()

    const handleClick = (val: string) =>
      // eslint-disable-next-line react-hooks/rules-of-hooks
      useCallback(
        (e: MouseEvent) => {
          e.preventDefault()
          onSelect(val)
        },
        [val],
      )

    useLayoutEffect(() => {
      if (droplistWidthAsInput && inputRef) {
        droplistRef.current.style.width = `${inputRef.current.offsetWidth}px`
      }
    }, [droplistWidthAsInput, inputRef])

    return (
      <StyledDroplist ref={mergeRefs([droplistRef, ref])} $isPanel={isPanel}>
        {isPanel && (
          <StyledDroplistHeader>
            <StyledDroplistTitle>{title}</StyledDroplistTitle>
            <StyledDroplistCancel onClick={close}>Отменить</StyledDroplistCancel>
          </StyledDroplistHeader>
        )}
        <Scrollable height={height || DROPLIST_SCROLL_DEFAULT_HEIGHT}>
          <StyledDroplistItems>
            {items.map((item) => {
              const checkmark = item.value === value ? <StyledChecked /> : null
              const content = (
                <>
                  {item.text}
                  {checkmark}
                </>
              )

              return (
                <StyledDroplistItem key={item.value} $size={size} onClick={handleClick(item.value)}>
                  {content}
                </StyledDroplistItem>
              )
            })}
          </StyledDroplistItems>
        </Scrollable>
      </StyledDroplist>
    )
  },
)

Droplist.displayName = 'Droplist'

export default Droplist
