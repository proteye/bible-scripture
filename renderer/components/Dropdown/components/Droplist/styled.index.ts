import { Checked } from '@dasreda/icons'
import { motion } from 'framer-motion'
import styled, { css } from 'styled-components'

import { IStyledDroplistItemProps, IStyledDroplistProps } from './types'

import mediaOld from 'components/media'

export const StyledChecked = styled(Checked)`
  ${({ theme }) => css`
    position: absolute;
    top: 50%;
    right: ${theme.spacings.s8};
    width: ${theme.spacings.s24};
    height: ${theme.spacings.s24};
    margin-top: 1px;
    transform: translateY(-50%);
    fill: ${theme.palette.emerald};
  `}
`

export const StyledDroplistItems = styled(motion.ul)`
  margin: -${({ theme }) => theme.spacings.s6} 0;
  padding: 0;
  overflow-y: hidden;
`

export const StyledDroplistItem = styled(motion.li)<IStyledDroplistItemProps>`
  ${({ theme, $size }) => css`
    position: relative;
    display: block;
    padding: ${theme.spacings.s6} 0;
    overflow: hidden !important;
    font: ${$size === 'small'
      ? theme.fontSystem.extraSmall
      : $size === 'medium'
      ? theme.fontSystem.small
      : theme.fontSystem.middle};
    white-space: nowrap;
    text-overflow: ellipsis;
    list-style: none;
    cursor: pointer;
    transition: color 0.3s ease-in-out;

    ${mediaOld.xl`
      &:hover {
        color: ${theme.palette.emerald};
      }
    `}
  `}
`

export const StyledDroplistHeader = styled.div`
  display: flex;
  justify-content: space-between;

  ${({ theme }) => css`
    padding: ${theme.spacings.s8} ${theme.spacings.s14} ${theme.spacings.s24} 0;
  `}
`

export const StyledDroplistTitle = styled.div`
  ${({ theme }) => css`
    color: ${theme.palette.textGraphite};
    font: ${theme.font.mobile.h2};
  `}
`

export const StyledDroplistCancel = styled.button`
  padding: 0;
  background-color: transparent;
  border: 0;

  ${({ theme }) => css`
    color: ${theme.palette.textGray};
    font: ${theme.font.mobile.p2};
  `}

  &:focus:not(:focus-visible) {
    outline: none;
  }
`

export const StyledDroplist = styled.div<IStyledDroplistProps>`
  ${({ theme, $isPanel }) => css`
    width: ${$isPanel ? '100%' : '348px'};
    padding: ${theme.spacings.s16} ${theme.spacings.s8} ${theme.spacings.s16}
      ${$isPanel ? theme.spacings.s24 : theme.spacings.s16};
    background-color: ${theme.palette.white};
    border-radius: ${theme.radius.r20};
    box-shadow: ${theme.palette.dropdownShadow};

    ${$isPanel &&
    css`
      border-bottom-right-radius: 0;
      border-bottom-left-radius: 0;
    `}
  `}
`
