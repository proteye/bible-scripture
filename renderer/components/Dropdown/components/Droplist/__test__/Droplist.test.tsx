import React from 'react'
import { cleanup, fireEvent } from '@testing-library/react'

import { JestReactTestingLibraryRenderWithTheme } from '../../../../../jest/JestReactTestingLibraryRenderWithTheme'
import mock from '../../../mock/cities.mock'
import Droplist from '../index'

describe('@dasreda/components/Droplist', () => {
  afterEach(cleanup)

  let onSelect = jest.fn()

  beforeEach(() => {
    onSelect = jest.fn()
  })

  test('Droplist: small', () => {
    const { container } = JestReactTestingLibraryRenderWithTheme(
      <Droplist items={mock.citiesLong} size="small" onSelect={onSelect} close={jest.fn} />,
    )

    expect(container).toMatchSnapshot()
  })

  test('Droplist: medium', () => {
    const { container } = JestReactTestingLibraryRenderWithTheme(
      <Droplist items={mock.citiesLong} size="medium" onSelect={onSelect} close={jest.fn} />,
    )

    expect(container).toMatchSnapshot()
  })

  test('Droplist: large', () => {
    const { container } = JestReactTestingLibraryRenderWithTheme(
      <Droplist items={mock.citiesLong} size="large" onSelect={onSelect} close={jest.fn} />,
    )

    expect(container).toMatchSnapshot()
  })

  test('Droplist: small with `value`', () => {
    const { container } = JestReactTestingLibraryRenderWithTheme(
      <Droplist
        items={mock.citiesLong}
        size="small"
        value={mock.citiesLong[0].value}
        onSelect={onSelect}
        close={jest.fn}
      />,
    )

    expect(container).toMatchSnapshot()
  })

  test('Droplist: medium with `onSelect`', () => {
    const { container, getAllByText } = JestReactTestingLibraryRenderWithTheme(
      <Droplist
        items={mock.citiesLong}
        size="medium"
        value={mock.citiesLong[0].value}
        onSelect={onSelect}
        close={jest.fn}
      />,
    )

    const someItem = getAllByText('Йошкар-Ола')[0]

    expect(someItem).toBeTruthy()

    fireEvent.click(someItem)

    expect(onSelect).toHaveBeenCalled()

    expect(container).toMatchSnapshot()
  })
})
