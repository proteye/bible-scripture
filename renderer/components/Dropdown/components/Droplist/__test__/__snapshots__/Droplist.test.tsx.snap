// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@dasreda/components/Droplist Droplist: large 1`] = `
.c1 {
  width: 100%;
}

.c1 .ScrollbarsCustom-Wrapper {
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.c1 .ScrollbarsCustom-Scroller {
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  margin-right: -21px;
  padding-right: 22px !important;
  overflow: hidden overlay !important;
}

.c1 .ScrollbarsCustom-Content {
  box-sizing: border-box;
  min-width: 100%;
  min-height: 100%;
}

.c1 .ScrollbarsCustom-TrackY {
  position: absolute;
  top: 0;
  right: 0px;
  width: 4px;
  height: 100%;
  overflow: hidden;
  background-color: transparent;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 .ScrollbarsCustom-ThumbY {
  top: 0;
  right: 8px;
  width: 4px;
  height: 56px;
  background-color: #D6D6DE;
  border-radius: 8px;
  -webkit-transform: translateY(0px);
  -ms-transform: translateY(0px);
  transform: translateY(0px);
  cursor: pointer;
}

.c2 {
  margin: -6px 0;
  padding: 0;
  overflow-y: hidden;
}

.c3 {
  position: relative;
  display: block;
  padding: 6px 0;
  overflow: hidden !important;
  font: 400 18px/26px "SB Sans Interface",sans-serif;
  white-space: nowrap;
  text-overflow: ellipsis;
  list-style: none;
  cursor: pointer;
  -webkit-transition: color 0.3s ease-in-out;
  transition: color 0.3s ease-in-out;
}

.c0 {
  width: 348px;
  padding: 16px 8px 16px 16px;
  background-color: #FFFFFF;
  border-radius: 20px;
  box-shadow: 0 6px 32px rgba(135,139,159,0.24);
}

@media (min-width:1280px) {
  .c3:hover {
    color: #21BA72;
  }
}

@media (min-width:1280px) {

}

@media only screen and (max-width:480px) {

}

@media only screen and (max-width:480px) {

}

<div>
  <div
    class="c0"
  >
    <div
      class="ScrollbarsCustom c1"
      style="position: relative; height: 250px;"
    >
      <div
        class="ScrollbarsCustom-Wrapper"
        style="position: absolute; overflow: hidden;"
      >
        <div
          class="ScrollbarsCustom-Scroller"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; overflow-y: hidden; overflow-x: hidden;"
        >
          <div
            class="ScrollbarsCustom-Content"
            style="padding: 0.05px;"
          >
            <ul
              class="c2"
            >
              <li
                class="c3"
              >
                Анапа - longlonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglong
              </li>
              <li
                class="c3"
              >
                Владмир
              </li>
              <li
                class="c3"
              >
                Волгоград
              </li>
              <li
                class="c3"
              >
                Димитров
              </li>
              <li
                class="c3"
              >
                Йошкар-Ола
              </li>
              <li
                class="c3"
              >
                Казань
              </li>
              <li
                class="c3"
              >
                Калининград
              </li>
              <li
                class="c3"
              >
                Киров
              </li>
              <li
                class="c3"
              >
                Липецк
              </li>
              <li
                class="c3"
              >
                Львов
              </li>
              <li
                class="c3"
              >
                Анапа
              </li>
              <li
                class="c3"
              >
                Владмир
              </li>
              <li
                class="c3"
              >
                Волгоград
              </li>
              <li
                class="c3"
              >
                Димитров
              </li>
              <li
                class="c3"
              >
                Йошкар-Ола
              </li>
              <li
                class="c3"
              >
                Казань
              </li>
              <li
                class="c3"
              >
                Калининград
              </li>
              <li
                class="c3"
              >
                Киров
              </li>
              <li
                class="c3"
              >
                Анапа
              </li>
              <li
                class="c3"
              >
                Владмир
              </li>
              <li
                class="c3"
              >
                Волгоград
              </li>
              <li
                class="c3"
              >
                Димитров
              </li>
              <li
                class="c3"
              >
                Йошкар-Ола
              </li>
              <li
                class="c3"
              >
                Казань
              </li>
              <li
                class="c3"
              >
                Калининград
              </li>
              <li
                class="c3"
              >
                Киров
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="ScrollbarsCustom-Track ScrollbarsCustom-TrackY"
        style="display: none;"
      >
        <div
          class="ScrollbarsCustom-Thumb ScrollbarsCustom-ThumbY"
        />
      </div>
      <div
        class="ScrollbarsCustom-Track ScrollbarsCustom-TrackX"
        style="display: none;"
      >
        <div
          class="ScrollbarsCustom-Thumb ScrollbarsCustom-ThumbX"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`@dasreda/components/Droplist Droplist: medium 1`] = `
.c1 {
  width: 100%;
}

.c1 .ScrollbarsCustom-Wrapper {
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.c1 .ScrollbarsCustom-Scroller {
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  margin-right: -21px;
  padding-right: 22px !important;
  overflow: hidden overlay !important;
}

.c1 .ScrollbarsCustom-Content {
  box-sizing: border-box;
  min-width: 100%;
  min-height: 100%;
}

.c1 .ScrollbarsCustom-TrackY {
  position: absolute;
  top: 0;
  right: 0px;
  width: 4px;
  height: 100%;
  overflow: hidden;
  background-color: transparent;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 .ScrollbarsCustom-ThumbY {
  top: 0;
  right: 8px;
  width: 4px;
  height: 56px;
  background-color: #D6D6DE;
  border-radius: 8px;
  -webkit-transform: translateY(0px);
  -ms-transform: translateY(0px);
  transform: translateY(0px);
  cursor: pointer;
}

.c2 {
  margin: -6px 0;
  padding: 0;
  overflow-y: hidden;
}

.c3 {
  position: relative;
  display: block;
  padding: 6px 0;
  overflow: hidden !important;
  font: 400 16px/24px "SB Sans Interface",sans-serif;
  white-space: nowrap;
  text-overflow: ellipsis;
  list-style: none;
  cursor: pointer;
  -webkit-transition: color 0.3s ease-in-out;
  transition: color 0.3s ease-in-out;
}

.c0 {
  width: 348px;
  padding: 16px 8px 16px 16px;
  background-color: #FFFFFF;
  border-radius: 20px;
  box-shadow: 0 6px 32px rgba(135,139,159,0.24);
}

@media (min-width:1280px) {
  .c3:hover {
    color: #21BA72;
  }
}

@media (min-width:1280px) {

}

@media only screen and (max-width:480px) {

}

@media only screen and (max-width:480px) {

}

<div>
  <div
    class="c0"
  >
    <div
      class="ScrollbarsCustom c1"
      style="position: relative; height: 250px;"
    >
      <div
        class="ScrollbarsCustom-Wrapper"
        style="position: absolute; overflow: hidden;"
      >
        <div
          class="ScrollbarsCustom-Scroller"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; overflow-y: hidden; overflow-x: hidden;"
        >
          <div
            class="ScrollbarsCustom-Content"
            style="padding: 0.05px;"
          >
            <ul
              class="c2"
            >
              <li
                class="c3"
              >
                Анапа - longlonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglong
              </li>
              <li
                class="c3"
              >
                Владмир
              </li>
              <li
                class="c3"
              >
                Волгоград
              </li>
              <li
                class="c3"
              >
                Димитров
              </li>
              <li
                class="c3"
              >
                Йошкар-Ола
              </li>
              <li
                class="c3"
              >
                Казань
              </li>
              <li
                class="c3"
              >
                Калининград
              </li>
              <li
                class="c3"
              >
                Киров
              </li>
              <li
                class="c3"
              >
                Липецк
              </li>
              <li
                class="c3"
              >
                Львов
              </li>
              <li
                class="c3"
              >
                Анапа
              </li>
              <li
                class="c3"
              >
                Владмир
              </li>
              <li
                class="c3"
              >
                Волгоград
              </li>
              <li
                class="c3"
              >
                Димитров
              </li>
              <li
                class="c3"
              >
                Йошкар-Ола
              </li>
              <li
                class="c3"
              >
                Казань
              </li>
              <li
                class="c3"
              >
                Калининград
              </li>
              <li
                class="c3"
              >
                Киров
              </li>
              <li
                class="c3"
              >
                Анапа
              </li>
              <li
                class="c3"
              >
                Владмир
              </li>
              <li
                class="c3"
              >
                Волгоград
              </li>
              <li
                class="c3"
              >
                Димитров
              </li>
              <li
                class="c3"
              >
                Йошкар-Ола
              </li>
              <li
                class="c3"
              >
                Казань
              </li>
              <li
                class="c3"
              >
                Калининград
              </li>
              <li
                class="c3"
              >
                Киров
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="ScrollbarsCustom-Track ScrollbarsCustom-TrackY"
        style="display: none;"
      >
        <div
          class="ScrollbarsCustom-Thumb ScrollbarsCustom-ThumbY"
        />
      </div>
      <div
        class="ScrollbarsCustom-Track ScrollbarsCustom-TrackX"
        style="display: none;"
      >
        <div
          class="ScrollbarsCustom-Thumb ScrollbarsCustom-ThumbX"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`@dasreda/components/Droplist Droplist: medium with \`onSelect\` 1`] = `
.c1 {
  width: 100%;
}

.c1 .ScrollbarsCustom-Wrapper {
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.c1 .ScrollbarsCustom-Scroller {
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  margin-right: -21px;
  padding-right: 22px !important;
  overflow: hidden overlay !important;
}

.c1 .ScrollbarsCustom-Content {
  box-sizing: border-box;
  min-width: 100%;
  min-height: 100%;
}

.c1 .ScrollbarsCustom-TrackY {
  position: absolute;
  top: 0;
  right: 0px;
  width: 4px;
  height: 100%;
  overflow: hidden;
  background-color: transparent;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 .ScrollbarsCustom-ThumbY {
  top: 0;
  right: 8px;
  width: 4px;
  height: 56px;
  background-color: #D6D6DE;
  border-radius: 8px;
  -webkit-transform: translateY(0px);
  -ms-transform: translateY(0px);
  transform: translateY(0px);
  cursor: pointer;
}

.c4 {
  position: absolute;
  top: 50%;
  right: 8px;
  width: 24px;
  height: 24px;
  margin-top: 1px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  fill: #21BA72;
}

.c2 {
  margin: -6px 0;
  padding: 0;
  overflow-y: hidden;
}

.c3 {
  position: relative;
  display: block;
  padding: 6px 0;
  overflow: hidden !important;
  font: 400 16px/24px "SB Sans Interface",sans-serif;
  white-space: nowrap;
  text-overflow: ellipsis;
  list-style: none;
  cursor: pointer;
  -webkit-transition: color 0.3s ease-in-out;
  transition: color 0.3s ease-in-out;
}

.c0 {
  width: 348px;
  padding: 16px 8px 16px 16px;
  background-color: #FFFFFF;
  border-radius: 20px;
  box-shadow: 0 6px 32px rgba(135,139,159,0.24);
}

@media (min-width:1280px) {
  .c3:hover {
    color: #21BA72;
  }
}

@media (min-width:1280px) {

}

@media only screen and (max-width:480px) {

}

@media only screen and (max-width:480px) {

}

<div>
  <div
    class="c0"
  >
    <div
      class="ScrollbarsCustom c1"
      style="position: relative; height: 250px;"
    >
      <div
        class="ScrollbarsCustom-Wrapper"
        style="position: absolute; overflow: hidden;"
      >
        <div
          class="ScrollbarsCustom-Scroller"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; overflow-y: hidden; overflow-x: hidden;"
        >
          <div
            class="ScrollbarsCustom-Content"
            style="padding: 0.05px;"
          >
            <ul
              class="c2"
            >
              <li
                class="c3"
              >
                Анапа - longlonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglong
                <svg
                  class="c4"
                  fill="none"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M7.05 11.576a1.248 1.248 0 00-1.682-.036 1.06 1.06 0 00-.04 1.57l3.567 3.546c.223.219.532.343.856.344h.004c.323 0 .63-.122.854-.339l8.056-7.777a1.06 1.06 0 00-.027-1.571 1.248 1.248 0 00-1.68.025L9.76 14.286l-2.712-2.71z"
                  />
                </svg>
              </li>
              <li
                class="c3"
              >
                Владмир
              </li>
              <li
                class="c3"
              >
                Волгоград
              </li>
              <li
                class="c3"
              >
                Димитров
              </li>
              <li
                class="c3"
              >
                Йошкар-Ола
              </li>
              <li
                class="c3"
              >
                Казань
              </li>
              <li
                class="c3"
              >
                Калининград
              </li>
              <li
                class="c3"
              >
                Киров
              </li>
              <li
                class="c3"
              >
                Липецк
              </li>
              <li
                class="c3"
              >
                Львов
              </li>
              <li
                class="c3"
              >
                Анапа
              </li>
              <li
                class="c3"
              >
                Владмир
              </li>
              <li
                class="c3"
              >
                Волгоград
              </li>
              <li
                class="c3"
              >
                Димитров
              </li>
              <li
                class="c3"
              >
                Йошкар-Ола
              </li>
              <li
                class="c3"
              >
                Казань
              </li>
              <li
                class="c3"
              >
                Калининград
              </li>
              <li
                class="c3"
              >
                Киров
              </li>
              <li
                class="c3"
              >
                Анапа
              </li>
              <li
                class="c3"
              >
                Владмир
              </li>
              <li
                class="c3"
              >
                Волгоград
              </li>
              <li
                class="c3"
              >
                Димитров
              </li>
              <li
                class="c3"
              >
                Йошкар-Ола
              </li>
              <li
                class="c3"
              >
                Казань
              </li>
              <li
                class="c3"
              >
                Калининград
              </li>
              <li
                class="c3"
              >
                Киров
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="ScrollbarsCustom-Track ScrollbarsCustom-TrackY"
        style="display: none;"
      >
        <div
          class="ScrollbarsCustom-Thumb ScrollbarsCustom-ThumbY"
        />
      </div>
      <div
        class="ScrollbarsCustom-Track ScrollbarsCustom-TrackX"
        style="display: none;"
      >
        <div
          class="ScrollbarsCustom-Thumb ScrollbarsCustom-ThumbX"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`@dasreda/components/Droplist Droplist: small 1`] = `
.c1 {
  width: 100%;
}

.c1 .ScrollbarsCustom-Wrapper {
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.c1 .ScrollbarsCustom-Scroller {
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  margin-right: -21px;
  padding-right: 22px !important;
  overflow: hidden overlay !important;
}

.c1 .ScrollbarsCustom-Content {
  box-sizing: border-box;
  min-width: 100%;
  min-height: 100%;
}

.c1 .ScrollbarsCustom-TrackY {
  position: absolute;
  top: 0;
  right: 0px;
  width: 4px;
  height: 100%;
  overflow: hidden;
  background-color: transparent;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 .ScrollbarsCustom-ThumbY {
  top: 0;
  right: 8px;
  width: 4px;
  height: 56px;
  background-color: #D6D6DE;
  border-radius: 8px;
  -webkit-transform: translateY(0px);
  -ms-transform: translateY(0px);
  transform: translateY(0px);
  cursor: pointer;
}

.c2 {
  margin: -6px 0;
  padding: 0;
  overflow-y: hidden;
}

.c3 {
  position: relative;
  display: block;
  padding: 6px 0;
  overflow: hidden !important;
  font: 400 14px/20px "SB Sans Interface",sans-serif;
  white-space: nowrap;
  text-overflow: ellipsis;
  list-style: none;
  cursor: pointer;
  -webkit-transition: color 0.3s ease-in-out;
  transition: color 0.3s ease-in-out;
}

.c0 {
  width: 348px;
  padding: 16px 8px 16px 16px;
  background-color: #FFFFFF;
  border-radius: 20px;
  box-shadow: 0 6px 32px rgba(135,139,159,0.24);
}

@media (min-width:1280px) {

}

@media only screen and (max-width:480px) {

}

@media only screen and (max-width:480px) {

}

@media (min-width:1280px) {
  .c3:hover {
    color: #21BA72;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="ScrollbarsCustom c1"
      style="position: relative; height: 250px;"
    >
      <div
        class="ScrollbarsCustom-Wrapper"
        style="position: absolute; overflow: hidden;"
      >
        <div
          class="ScrollbarsCustom-Scroller"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; overflow-y: hidden; overflow-x: hidden;"
        >
          <div
            class="ScrollbarsCustom-Content"
            style="padding: 0.05px;"
          >
            <ul
              class="c2"
            >
              <li
                class="c3"
              >
                Анапа - longlonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglong
              </li>
              <li
                class="c3"
              >
                Владмир
              </li>
              <li
                class="c3"
              >
                Волгоград
              </li>
              <li
                class="c3"
              >
                Димитров
              </li>
              <li
                class="c3"
              >
                Йошкар-Ола
              </li>
              <li
                class="c3"
              >
                Казань
              </li>
              <li
                class="c3"
              >
                Калининград
              </li>
              <li
                class="c3"
              >
                Киров
              </li>
              <li
                class="c3"
              >
                Липецк
              </li>
              <li
                class="c3"
              >
                Львов
              </li>
              <li
                class="c3"
              >
                Анапа
              </li>
              <li
                class="c3"
              >
                Владмир
              </li>
              <li
                class="c3"
              >
                Волгоград
              </li>
              <li
                class="c3"
              >
                Димитров
              </li>
              <li
                class="c3"
              >
                Йошкар-Ола
              </li>
              <li
                class="c3"
              >
                Казань
              </li>
              <li
                class="c3"
              >
                Калининград
              </li>
              <li
                class="c3"
              >
                Киров
              </li>
              <li
                class="c3"
              >
                Анапа
              </li>
              <li
                class="c3"
              >
                Владмир
              </li>
              <li
                class="c3"
              >
                Волгоград
              </li>
              <li
                class="c3"
              >
                Димитров
              </li>
              <li
                class="c3"
              >
                Йошкар-Ола
              </li>
              <li
                class="c3"
              >
                Казань
              </li>
              <li
                class="c3"
              >
                Калининград
              </li>
              <li
                class="c3"
              >
                Киров
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="ScrollbarsCustom-Track ScrollbarsCustom-TrackY"
        style="display: none;"
      >
        <div
          class="ScrollbarsCustom-Thumb ScrollbarsCustom-ThumbY"
        />
      </div>
      <div
        class="ScrollbarsCustom-Track ScrollbarsCustom-TrackX"
        style="display: none;"
      >
        <div
          class="ScrollbarsCustom-Thumb ScrollbarsCustom-ThumbX"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`@dasreda/components/Droplist Droplist: small with \`value\` 1`] = `
.c1 {
  width: 100%;
}

.c1 .ScrollbarsCustom-Wrapper {
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.c1 .ScrollbarsCustom-Scroller {
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  margin-right: -21px;
  padding-right: 22px !important;
  overflow: hidden overlay !important;
}

.c1 .ScrollbarsCustom-Content {
  box-sizing: border-box;
  min-width: 100%;
  min-height: 100%;
}

.c1 .ScrollbarsCustom-TrackY {
  position: absolute;
  top: 0;
  right: 0px;
  width: 4px;
  height: 100%;
  overflow: hidden;
  background-color: transparent;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 .ScrollbarsCustom-ThumbY {
  top: 0;
  right: 8px;
  width: 4px;
  height: 56px;
  background-color: #D6D6DE;
  border-radius: 8px;
  -webkit-transform: translateY(0px);
  -ms-transform: translateY(0px);
  transform: translateY(0px);
  cursor: pointer;
}

.c4 {
  position: absolute;
  top: 50%;
  right: 8px;
  width: 24px;
  height: 24px;
  margin-top: 1px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  fill: #21BA72;
}

.c2 {
  margin: -6px 0;
  padding: 0;
  overflow-y: hidden;
}

.c3 {
  position: relative;
  display: block;
  padding: 6px 0;
  overflow: hidden !important;
  font: 400 14px/20px "SB Sans Interface",sans-serif;
  white-space: nowrap;
  text-overflow: ellipsis;
  list-style: none;
  cursor: pointer;
  -webkit-transition: color 0.3s ease-in-out;
  transition: color 0.3s ease-in-out;
}

.c0 {
  width: 348px;
  padding: 16px 8px 16px 16px;
  background-color: #FFFFFF;
  border-radius: 20px;
  box-shadow: 0 6px 32px rgba(135,139,159,0.24);
}

@media (min-width:1280px) {
  .c3:hover {
    color: #21BA72;
  }
}

@media (min-width:1280px) {

}

@media only screen and (max-width:480px) {

}

@media only screen and (max-width:480px) {

}

<div>
  <div
    class="c0"
  >
    <div
      class="ScrollbarsCustom c1"
      style="position: relative; height: 250px;"
    >
      <div
        class="ScrollbarsCustom-Wrapper"
        style="position: absolute; overflow: hidden;"
      >
        <div
          class="ScrollbarsCustom-Scroller"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; overflow-y: hidden; overflow-x: hidden;"
        >
          <div
            class="ScrollbarsCustom-Content"
            style="padding: 0.05px;"
          >
            <ul
              class="c2"
            >
              <li
                class="c3"
              >
                Анапа - longlonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglong
                <svg
                  class="c4"
                  fill="none"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M7.05 11.576a1.248 1.248 0 00-1.682-.036 1.06 1.06 0 00-.04 1.57l3.567 3.546c.223.219.532.343.856.344h.004c.323 0 .63-.122.854-.339l8.056-7.777a1.06 1.06 0 00-.027-1.571 1.248 1.248 0 00-1.68.025L9.76 14.286l-2.712-2.71z"
                  />
                </svg>
              </li>
              <li
                class="c3"
              >
                Владмир
              </li>
              <li
                class="c3"
              >
                Волгоград
              </li>
              <li
                class="c3"
              >
                Димитров
              </li>
              <li
                class="c3"
              >
                Йошкар-Ола
              </li>
              <li
                class="c3"
              >
                Казань
              </li>
              <li
                class="c3"
              >
                Калининград
              </li>
              <li
                class="c3"
              >
                Киров
              </li>
              <li
                class="c3"
              >
                Липецк
              </li>
              <li
                class="c3"
              >
                Львов
              </li>
              <li
                class="c3"
              >
                Анапа
              </li>
              <li
                class="c3"
              >
                Владмир
              </li>
              <li
                class="c3"
              >
                Волгоград
              </li>
              <li
                class="c3"
              >
                Димитров
              </li>
              <li
                class="c3"
              >
                Йошкар-Ола
              </li>
              <li
                class="c3"
              >
                Казань
              </li>
              <li
                class="c3"
              >
                Калининград
              </li>
              <li
                class="c3"
              >
                Киров
              </li>
              <li
                class="c3"
              >
                Анапа
              </li>
              <li
                class="c3"
              >
                Владмир
              </li>
              <li
                class="c3"
              >
                Волгоград
              </li>
              <li
                class="c3"
              >
                Димитров
              </li>
              <li
                class="c3"
              >
                Йошкар-Ола
              </li>
              <li
                class="c3"
              >
                Казань
              </li>
              <li
                class="c3"
              >
                Калининград
              </li>
              <li
                class="c3"
              >
                Киров
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="ScrollbarsCustom-Track ScrollbarsCustom-TrackY"
        style="display: none;"
      >
        <div
          class="ScrollbarsCustom-Thumb ScrollbarsCustom-ThumbY"
        />
      </div>
      <div
        class="ScrollbarsCustom-Track ScrollbarsCustom-TrackX"
        style="display: none;"
      >
        <div
          class="ScrollbarsCustom-Thumb ScrollbarsCustom-ThumbX"
        />
      </div>
    </div>
  </div>
</div>
`;
